onnxruntime/core/providers/dnnl/dnnl_execution_provider.cc

```diff
@@ -5,42 +5,51 @@
 #pragma warning(disable : 4996)
 #endif
 
-#include "core/framework/allocator.h"
-#include "core/framework/compute_capability.h"
-#include "core/framework/kernel_registry.h"
-#include "core/providers/dnnl/subgraph/dnnl_func_kernel.h"
+#include "core/providers/shared_library/provider_author.h"
+#include <unordered_set>
+#include "subgraph/dnnl_func_kernel.h"
 #include "dnnl_execution_provider.h"
 #include "dnnl_fwd.h"
 
-namespace {
+const OrtApi* ORT_API_CALL GetApi(uint32_t /*version*/) NO_EXCEPTION { return nullptr; }
+const char* ORT_API_CALL GetVersionString() NO_EXCEPTION { return "invalid"; }
+
+static OrtApiBase ort_api_base = {
+    &GetApi,
+    &GetVersionString,
+};
+
+const OrtApiBase* ORT_API_CALL OrtGetApiBase() NO_EXCEPTION {
+  return &ort_api_base;
+}
+
+namespace onnxruntime {
+
 struct KernelRegistryAndStatus {
-  std::shared_ptr<onnxruntime::KernelRegistry> kernel_registry = std::make_shared<onnxruntime::KernelRegistry>();
+  std::shared_ptr<onnxruntime::Prov_KernelRegistry> kernel_registry{Prov_KernelRegistry::Create()};
```

Comment:
naming convention: ProvKernelRegistry, or better ProviderKernelRegistry. Same for all places where you've underscore separated class names.

Reply:
Sounds good, I also was thinking of having a Provider namespace but that might be confusing too.